pkg/scheduler/factory/factory_test.go

```diff
@@ -252,48 +253,126 @@ func TestDefaultErrorFunc(t *testing.T) {
 	client := fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testPod}})
 	stopCh := make(chan struct{})
 	defer close(stopCh)
-	queue := &internalqueue.FIFO{FIFO: cache.NewFIFO(cache.MetaNamespaceKeyFunc)}
+
+	var (
+		timestamp    = time.Now()
+		foundPodFlag = false
+	)
+	queue := internalqueue.NewPriorityQueueWithClock(nil, clock.NewFakeClock(timestamp))
 	schedulerCache := internalcache.New(30*time.Second, stopCh)
-	podBackoff := internalqueue.NewPodBackoffMap(1*time.Second, 60*time.Second)
-	errFunc := MakeDefaultErrorFunc(client, podBackoff, queue, schedulerCache, stopCh)
+	errFunc := MakeDefaultErrorFunc(client, queue, schedulerCache, stopCh)
 
+	// Trigger error handling again to put the pod in unschedulable queue
 	errFunc(testPod, nil)
 
-	for {
-		// This is a terrible way to do this but I plan on replacing this
-		// whole error handling system in the future. The test will time
-		// out if something doesn't work.
+	// Try up to a second to retrieve the error pod from priority queue
+	for i := 0; i < 100; i++ {
 		time.Sleep(10 * time.Millisecond)
-		got, exists, _ := queue.Get(testPod)
+		got, exists := getPodfromPriorityQueue(queue, testPod)
 		if !exists {
 			continue
 		}
-		requestReceived := false
-		actions := client.Actions()
-		for _, a := range actions {
-			if a.GetVerb() == "get" {
-				getAction, ok := a.(clienttesting.GetAction)
-				if !ok {
-					t.Errorf("Can't cast action object to GetAction interface")
-					break
-				}
-				name := getAction.GetName()
-				ns := a.GetNamespace()
-				if name != "foo" || ns != "bar" {
-					t.Errorf("Expected name %s namespace %s, got %s %s",
-						"foo", "bar", name, ns)
-				}
-				requestReceived = true
-			}
+
+		testClientGetPodRequest(client, t)
+
+		if e, a := testPod, got; !reflect.DeepEqual(e, a) {
+			t.Errorf("Expected %v, got %v", e, a)
 		}
-		if !requestReceived {
-			t.Errorf("Get pod request not received")
+
+		foundPodFlag = true
+		break
+	}
+
+	if !foundPodFlag {
+		t.Errorf("Failed to get pod from the unschedulable queue after waiting for a minute: %v", testPod)
```

Comment:
s/minute/second/

1 second may be too short in some of our test clusters. You may want to increase the number of iterations of the loop or the sleep duration.

Reply:
My bad. I meant to loop up to a minute. Now I increased the sleep duration to 100ms and the number of iterations to 600.